docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,598 @@
+```
+
+As you can see, requests to the user data source were batched, thus fetching all the data in one round.
+
+### Traverse
+
+Another interesing combinator is `traverse`, which is the composition of `map` and `sequence`.
+
+```scala
+val fch: Fetch[List[User]] = List(1, 2, 3).traverse(getUser)
+
+val result: List[User] = Fetch.run[Id](fch)
+// Fetching users List(1, 2, 3)
+
+//=> result: List[User] = List(User(1,@egg_1), User(2,@egg_2), User(3,@egg_3))
+```
+
+# Interpreting a fetch to a concurrency monad
+
+Albeit the examples use `Id` as the concurrency Monad, `Fetch` is not limited to just `Id`, any monad `M` that
+implements `MonadError[M, Throwable]` will do. Fetch provides `MonadError` instances for some existing monads like
+`Future`, `cats.Id` and `cats.Eval` and is easy to write your own.
+
+In practice you'll be interpreting a fetch to a concurrency monad like `Future` or `scalaz.concurrent.Task` to exploit
+parallelism whenever we can make requests to multiple independent data sources at the same time.
+
+## Future
+
+For interpreting a fetch into a `Future` we must first import the `MonadError[Future, Throwable]` available in cats.
+
+```scala
+import cats.std.future._
+```
+
+We can now interpret a fetch into a future:
+
+```scala
+import cats.std.future._
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+val fch: Fetch[(User, Post)] = (getUser(1) |@| getPost(1)).tupled
+
+val fut: Future[(User, Post)] = Fetch.run[Future](fch)
+// Fetching users List(1)
+// Fetching posts List(1)
+
+Await.result(fut, 1 seconds)
+//=> (User, Post) = (User(1,@egg_1),Post(1,2,An article with id 1))
+```
+
+# Caching
+
+As we have learned, Fetch caches intermediate results implicitly using a cache. You can
+provide a prepopulated cache for running a fetch, replay a fetch with the cache of a previous
+one and even implement a custom cache.
+
+## Prepopulating a cache
+
+We'll be using the default in-memory cache, prepopulated with some data. The cache key of an identity
+is calculated with the `DataSource`'s `identity` method; we can pass a cache as the second argument when
+running a fetch with `Fetch.run`.
+
+```scala
+val fch: Fetch[User] = getUser(1)
+val cache = InMemoryCache(UserSource.identity(1) -> User(1, "@dialelo"))
+
+val result: User = Fetch.run[Id](fch, cache)
+//=> result: User = User(1,@dialelo)
+```
+
+As you can see, when all the data is cached no fetch is executed at all since the results are available
+in the cache.
+
+If only part of the data is cached, the cached data won't be asked for:
+
+```scala
+val fch: Fetch[List[User]] = List(1, 2, 3).traverse(getUser)
+val cache = InMemoryCache(UserSource.identity(2) -> User(2, "@two"))
+
+val result: List[User] = Fetch.run[Id](fch, cache)
+// Fetching users List(1, 3)
+
+//=> result: List[User] = List(User(1,@egg_1), User(2,@two), User(3,@egg_3))
+```
+
+## Replaying a fetch
```

Comment:
This is misleading because Fetches can be replayed regardless of caching but this section talks more about how you can reuse a Fetch cache